varcode/variant_collection.py

```diff
@@ -110,49 +67,55 @@ def __str__(self):
     def __repr__(self):
         return str(self)
 
-    def clone(self, new_variants=None):
+    def clone_metadata(self, new_variants):
```

Comment:
i'd suggest having one method called `clone` defined immediately before or after __init__ (so it's easy to keep the arguments in agreement) that takes exactly the same arguments as __init__ but with all arguments optional. Any arguments specified to that function will be used in the returned VariantCollection, and any arguments omitted will default to the values in the current instance. Otherwise having multiple clone methods and keeping track of what's considered "metadata" as new fields are added in the future seems confusing to me